front/src/pages/onboarding/register.jsx

```diff
@@ -0,0 +1,83 @@
+import Page from "@/components/Page"
+import SubmitButton from "@/components/ui/buttons/SubmitButton"
+import FormField from "@/components/ui/forms/FormField"
+import MyForm from "@/components/ui/forms/MyForm"
+import { createUser } from "@/utils/services/userServices"
+import routes from "@@/lib/constants/routes"
+import { emailValidator, firstNameValidator, lastNameValidator, passwordValidator } from "@@/lib/validators/user"
+import { Formik } from "formik"
+import { serverSideTranslations } from "next-i18next/serverSideTranslations"
+import Link from "next/link"
+import { useRouter } from "next/router"
+import { useState } from "react"
+import { useTranslation } from "react-i18next"
+import { object } from "yup"
+// The only way to have translations work on this page was to disable SSR
+import NoSSR from "@/components/NoSSR"
+
+export const getServerSideProps = async ({ locale }) => ({ props: { ...(await serverSideTranslations(locale, ["common"])) } })
+const initialValues = {
+  firstName: "",
+  lastName: "",
+  email: "",
+  password: ""
+}
+
+const validationSchema = object({
+  email: emailValidator.required(),
+  password: passwordValidator.required(),
+  firstName: firstNameValidator.required(),
+  lastName: lastNameValidator.required()
+})
+
+const RegisterPage = () => {
+  const { t } = useTranslation("common")
+  const router = useRouter()
+  const [hasError, setError] = useState(false)
+  const handleUserCreation = async values => {
+    const { firstName, lastName, email, password } = values
+
+    try {
+      const user = { firstName, lastName, email, password }
+      await createUser(user)
+      router.push(routes.home)
+    } catch (e) {
+      setError(true)
+    }
+  }
+
+  if (hasError) {
+    return <p className="bg-red-700">{t("errorMessage")}</p>
+  }
+
+  return (
+    <>
+      <Page className="flex justify-center">
+        <NoSSR>
+          <div className="flex flex-col justify-center max-w-96">
+            <h1 className="text-center text-3xl rounded-md">{t("auth.register")}</h1>
```

Comment:
It's about time to understand that React is all about reusable components.

![](https://cdn.frankerfacez.com/emoticon/369539/4)